optuna/storages/_cached_storage.py

```diff
@@ -199,6 +200,11 @@ def set_trial_state(self, trial_id: int, state: TrialState) -> bool:
                 updates = self._get_updates(trial_id)
                 cached_trial.state = state
                 updates.state = state
+
+                if state == TrialState.RUNNING:
+                    updates.datetime_start = datetime.datetime.now()
+                    cached_trial.datetime_start = datetime.datetime.now()
```

Comment:
If we set a running trial to running once again, it seems like `datetime_start` will be updated. I think it's inconsistent with the other storages. How about only updating the timestamp here if the state actually changes? I'm sorry for the lack of missing tests here.

Reply:
Agreed. Will add that in next commit.